### PR TITLE
post: cool things we learned frontend edition

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+Basically, in a CLI project, you run `ng add @angular/pwa`, and... You're done!
+You'll need to customize the icons and colors to replace the default ones,
+and the CLI will then generate all the required files for a proper PWA.
+With this default setup, and the application served over HTTPS,
+our user are offered the possibility to add the application to their home screen.
+This is not the only perk:
+the application will also start very fast
+as the assets are cached for offline use.
+
+We tried to push things a little bit more and used the [`SwUpdate`](https://angular.io/api/service-worker/SwUpdate) service offered by the `@angular/pwa` package to display a notification to our users when a new version of the applciation is available.
```
Comment (jnizet) [inline _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md]: a little further?
applciation -> application

File: _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+This is because the application is displayed right away from cache when a user loads it, and then the Service Worker checks if a new version is available.
+If that's the case, [`SwUpdate.available`](https://angular.io/api/service-worker/SwUpdate#properties) emits a new event,
+and you can ask your user if he/she wants to refresh the page to use the new version.
+This works great even we ran into an Angular issue, that [we documented](https://github.com/angular/angular/commit/353362f5a4ce14e91cc96359a990013f10747b47) as good open source citizens 😉
```
Comment (jnizet) [inline _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md]: I'd rephrase to something like this:
> We had a weird issue, but we figured it out, and, as good open source citizens, we documented it.

File: _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## Leaflet and ngx-leaflet
+
+We needed to display a map in several places in the application,
+with pins for location of the issues reported for example.
```
Comment (jnizet) [inline _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md]: with pins marking the locations of the reported issues.

File: _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+Our customer really wanted to use Open Street Map,
+so Google Maps was not an option.
+We went with [Leaflet](https://leafletjs.com/) and found [ngx-leaflet](https://github.com/Asymmetrik/ngx-leaflet),
+a nice little library offering an Angular directive.
+It proved super easy and straight-forward to use,
+and we did not regret to use it!
```
Comment (jnizet) [inline _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md]: I would add a personal kudos to the guy who creates and maintains it.

File: _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md (hunk 1 above)
Comment (clacote) [inline _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md]: applciation

File: _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md (hunk 2 above)
Comment (clacote) [inline _posts/2019-03-14-cool-things-we-learned-part-2-frontend-edition.md]: "This works great even we ran" sound incorrect, but maybe that's me. Don't you want to say "This works great, even if we ran"?

**Comment by cexbrayat** [comment]
Thanks for the reviews, I fixed, and I'll publish tomorrow morning.
